WalletWasabi/WabiSabi/Backend/Rounds/Arena.cs

```diff
@@ -559,16 +561,16 @@ private void TimeoutAlices()
 		}
 	}
 
-	private async Task<ConstructionState> TryAddBlameScriptAsync(Round round, ConstructionState coinjoin, bool allReady, Script blameScript, CancellationToken cancellationToken)
+	internal static async Task<ConstructionState> TryAddBlameScriptAsync(Round round, ConstructionState coinjoin, bool allReady, Script blameScript, Func<Task<FeeRate>> highestFeeRateAsyncMethod, CancellationToken cancellationToken)
 	{
 		// If timeout we must fill up the outputs to build a reasonable transaction.
 		// This won't be signed by the alice who failed to provide output, so we know who to ban.
-		var estimatedBlameScriptCost = round.Parameters.MiningFeeRate.GetFee(blameScript.EstimateOutputVsize() + coinjoin.UnpaidSharedOverhead);
-		var diffMoney = coinjoin.Balance - coinjoin.EstimatedCost - estimatedBlameScriptCost;
+		var diffMoney = Money.Satoshis(round.Parameters.MiningFeeRate.SatoshiPerByte * (coinjoin.EstimatedVsize - coinjoin.UnpaidSharedOverhead - blameScript.EstimateOutputVsize()));
```

Comment:
This calculation is not correct. The test showed it is giving back dust amounts in some cases.

Reply:
Fixed the calculation.